internal/provider/resource_gitlab_group_access_token.go

```diff
@@ -0,0 +1,236 @@
+	"developer",
+	"maintainer",
+}
+
+var _ = registerResource("gitlab_group_access_token", func() *schema.Resource {
+	// NOTE: importing a token is not possible, because it is only exposed during creating by the GitLab API.
+	// lintignore: XR002 // NOTE: importing a token is not possible, because it is only exposed during creating by the GitLab API.
+	return &schema.Resource{
+		Description: "This resource allows you to create and manage Group Access Token for your GitLab Groups. (Introduced in GitLab 14.7)",
+
+		CreateContext: resourceGitlabGroupAccessTokenCreate,
+		ReadContext:   resourceGitlabGroupAccessTokenRead,
+		DeleteContext: resourceGitlabGroupAccessTokenDelete,
+
+		Schema: map[string]*schema.Schema{
+			"group": {
+				Description: "The ID or path of the group to add the group access token to.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"name": {
+				Description: "The name of the group access token.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"scopes": {
+				Description: fmt.Sprintf("The scope for the group access token. It determines the actions which can be performed when authenticating with this token. Valid values are: %s.", renderValueListForDocs(validGroupAccessTokenScopes)),
+				Type:        schema.TypeSet,
+				Required:    true,
+				ForceNew:    true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringInSlice(validGroupAccessTokenScopes, false),
+				},
+			},
+			"access_level": {
+				Description:      fmt.Sprintf("The access level for the group access token. Valid values are: %s.", renderValueListForDocs(validAccessLevels)),
+				Type:             schema.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				Default:          accessLevelValueToName[gitlab.MaintainerPermissions],
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(validAccessLevels, false)),
+			},
+			"expires_at": {
+				Description: "The token expires at midnight UTC on that date. The date must be in the format YYYY-MM-DD. Default is never.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				ValidateDiagFunc: func(i interface{}, p cty.Path) diag.Diagnostics {
+					v := i.(string)
+
+					if _, err := time.Parse("2006-01-02", v); err != nil {
+						return diag.Errorf("expected %q to be a valid YYYY-MM-DD date, got %q: %+v", p, i, err)
+					}
+
+					return nil
+				},
+			},
+			"token": {
+				Description: "The group access token. This is only populated when creating a new group access token.",
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+			},
+			"active": {
+				Description: "True if the token is active.",
+				Type:        schema.TypeBool,
+				Computed:    true,
+			},
+			"created_at": {
+				Description: "Time the token has been created, RFC3339 format.",
```

Comment:
That's a lie. The code is setting `created_at` by calling `String()` on the time, which uses `"2006-01-02 15:04:05.999999999 -0700 MST"` format.

For now and in the future I think we should always format the time attributes according to how they appear in the GitLab API. Basically, go-gitlab parses them into `Time` objects, and we format them back, using `time.Format()` instead of `String()`. :)

Reply:
good catch!